### DISCUSSION TITLE
DietPi AutoStart to Chromium, laggy video when adding ticker activity in Pixi.js app; fine on MBP

### DISCUSSION THREAD

**Discussion by karts-with-freaking-lasers** [pixijs/pixijs#8260] [Q&A]
Please forgive my brevity, this is such a loaded question. I hope this connects enough dots to begin a healthy discussion. I'm happy to add details where needed--just wanted to frame the question as best I could.

On a MacBook Pro 2015, this same setup with `THE_LINE_IN_QUESTION` uncommented allows the video to run smoothly.

**Raspberry Pi 4**:
 - DietPi: AutoStart w/ Chromium Kiosk Mode opening a locally hosted Pixi.js app
 - Chromium has Hardware Acceleration *enabled*, which is a requirement to play smooth 1080p (But, I'm actually using a 720p video because it's playing on a rather small screen and for to help with the limited hardware vs. my MBP).

**Pixi.js App**:

index.html:
```html
  <canvas id="pixiApp"></canvas>
  <video id="myVideo" src="myVideo.mp4"></video>
```
Inside `pixiApp`, if I comment out `THE_LINE_IN_QUESTION`...
```javascript
  const app = new PIXI.Application({
    view: pixiApp
  });
  engine.update = function noop() {};
  app.ticker.stop(); // <= Stopped, so gsap can control ticking
  app.ticker.add(engine.update, engine);
  gsap.ticker.add((delta, ...) => {
    // This adds the `app` to the `gsap` update loop; commented out,
    // it allows the video to play smoothly but no animations or 
    // transitioning within my `engine`. I could temporarily suspend 
    // the ticker when the video should play--and that's my fallback plan.
    // But I wanted to understand if there was a shortfall of mine 
    // that could make the FPS issue go away with the following uncommented.
    // A shift in paradigm away from "web development" (which is my background)
    // into "gaming" with update loops and such.
    // THE_LINE_IN_QUESTION:
    app.ticker.update(delta); // <= Makes video choppy on rPi4 (not MBP), even with a `noop`--though, I'm sure there is other stuff happening inside Pixi.js.
  });
```
...then `myVideo` plays smoothly. If, however, I go through the effort of [playing my video through pixijs](https://www.html5gamedevs.com/topic/44283-play-video-in-pixi-v5/), FPS is reduced to a stall, or choppy at best.

Why does `THE_LINE_IN_QUESTION` add such choppiness to the Pi4 where it runs fine on a MacBook Pro?

Edit:
Just [playing video through pixijs](https://www.html5gamedevs.com/topic/44283-play-video-in-pixi-v5/), FPS stalls, or is choppy, on the Pi4 (not MBP). I then added the `<video>` tag so PixiJs wasn't "processing" the video but then with the Pixi ticker running along with the video, the video is very choppy. Hopefully this helps clarify.

**Comment by andrewstart** [comment]
Every time I've combined video and PixiJS in the past, I have found that running the video through PixiJS just did not perform well enough on weaker devices (mostly mobile ones), and got better performance out of having a separate video element behind the transparent PixiJS canvas. It sounds like you are doing that but still finding poor performance on the Pi4 - in that case my assumption is that the Pi4 doesn't have the resources to do both video processing and a WebGL canvas at the same time (maybe only one can be hardware accelerated at a time, or some other limitation?).

**Reply by karts-with-freaking-lasers** [reply]
Ah, that is not something I had considered! I'm glad I ventured out.

@andrewstart , I appreciate your perspective leading to a sound advice: I shall continue with my `fallback plan` for now and investigate the `Pi4 + Hardware Acceleration + WebGL(Canvas + Video) = Choppy Video` equation when I have time again in 6-12 months.

Cheers!

Edit:
Here is some relevant data--anything to be adjusted here to help?
<img width="803" alt="image" src="https://user-images.githubusercontent.com/93396860/161632320-3585eaba-3700-4efb-a6d3-eab25557bcbb.png">
<img width="607" alt="image" src="https://user-images.githubusercontent.com/93396860/161632433-3fb84a0b-cda0-4a28-8df3-5ec1daa78e4a.png">
